app.py

```diff
@@ -71,986 +100,1068 @@
+		if votes is None:
+			db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("votes").child(str(index)).update({0: user})
+			current = db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("message").child(str(index)).child("text").child("text").get().val()
+			db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("message").child(str(index)).child("text").update({"text": current.strip() + " `1` ~ <@" + user + ">"})
+		else:
+			current = db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("message").child(str(index)).child("text").child("text").get().val()
+			if user not in current:
+				i = len(votes)
+				db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("votes").child(str(index)).update({i: user})
+				db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("message").child(str(index)).child("text").update({"text": current.split("`")[0] + "`" + str(i + 1) + "` ~ <@" + user + ">" + current.split("~")[1]})
+			else:
+				for i in votes:
+					if i == user:
+						votes.remove(i)
+				db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("votes").child(str(index)).remove()
+				j = 0
+				for i in votes:
+					db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("votes").child(str(index)).update({j: i})
+					j += 1
+				current = db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("message").child(str(index)).child("text").child("text").get().val()
+				if j == 0:
+					db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("message").child(str(index)).child("text").update({"text": current.split("`")[0]})
+				else:
+					db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).child("message").child(str(index)).child("text").update({"text": current.split("`")[0] + "`" + str(j) + "` ~ " + current.split("~")[1].replace("<@" + user + ">", "").strip()})
+		blocks = dict(db.child(key_fb_tars).child("polls").child(ts.replace(".", "-")).get().val())["message"]
+		tars.chat_update(channel=channel, ts=ts, blocks=blocks)
+
+@app.route("/login", methods=["GET", "POST"])
+def login():
+	if(request.method == "GET"):
+		data1={}
+		data1["status"]="Enter Credentials"
+		print("Sending login page")
+		return render_template("login.html", data=data1)
+	
+	if(request.method=="POST"):
+		username_form=request.form.get("username")
+		password_form=request.form.get("password").encode()
+		if(username_form==username and bcrypt.checkpw(password_form, password)):
+			user = User()
+			user.id = username
+			flask_login.login_user(user)
+			data1={}
+			data1["status"] = "Enter the above details and submit"
+			return redirect(request.args.get("next") or url_for("index"))
+		else:
+			data1={}
+			data1["status"]="Credentials incorrect"
+			return render_template("login.html",data=data1)
+
+@app.route("/addpublication",methods=["GET","POST"])
```

Comment:
```suggestion
@app.route("/add-publication", methods=["GET", "POST"])
```
`add-publication` seems better?